File: src/main/java/be/lutske/leolegacy/infrastructure/persistence/entity/CategoryEntity.java
package be.lutske.leolegacy.infrastructure.persistence.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "categories")
public class CategoryEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    @Column(nullable = false, unique = true, length = 100)
    public String slug;

    @Column(nullable = false, unique = true, length = 100)
    public String name;

    @Column(name = "display_order", nullable = false)
    public Integer displayOrder;
}

File: backend/src/main/kotlin/be/lutske/leolegacy/infrastructure/persistence/entity/CategoryEntity.kt
package be.lutske.leolegacy.infrastructure.persistence.entity

import io.quarkus.hibernate.orm.panache.kotlin.PanacheEntityBase
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "category")
class CategoryEntity : PanacheEntityBase {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0

    lateinit var name: String

    constructor()

    constructor(id: Long, name: String) {
        this.id = id
        this.name = name
    }
}
